python/pyspark/ml/torch/distributor.py

```diff
@@ -0,0 +1,120 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from .utils import get_conf_boolean
+import math
+from typing import Union, Callable
+from pyspark.sql import SparkSession
+
+
+# might need to move into its own file as we look forward.
+class Distributor:
+    def __init__(self, num_processes: int = 1, local_mode: bool = True, use_gpu: bool = True):
+        self.num_processes = num_processes
+        self.local_mode = local_mode
+        self.use_gpu = use_gpu
+        self.spark = SparkSession.builder.getOrCreate()
+        self.sc = self.spark.sparkContext
+        self.num_tasks = self._get_num_tasks()
+        
+    def _get_num_tasks(self):
+        """
+        Returns:
+            The number of Spark tasks to use for distributed training
+        """
+        if self.use_gpu:
+            key = "spark.task.resource.gpu.amount"
+            if self.sc.getConf().contains(key):
+                task_gpu_amount = int(self.sc.getConf().get(key))
+            else:
+                task_gpu_amount = 1 # for single node clusters
+            if task_gpu_amount < 1:
+                raise ValueError(
+                    f"The Spark conf `{key}` has a value "
+                    f"of {task_gpu_amount} but it "
+                    "should not have a value less than 1."
+                )
+            return math.ceil(self.num_processes / task_gpu_amount)
+        return self.num_processes
+    
+    def _validate_input_params(self):
+        if self.num_processes <= 0:
+            raise ValueError(f"num_proccesses has to be a positive integer")
+
+    # can be required for TF distributor in the future as well
+    def _check_encryption(self):
+        if "ssl_conf" not in self.__dict__:
+            raise NotImplementedError()
+        is_ssl_enabled = get_conf_boolean(
+            self.sc,
+            "spark.ssl.enabled",
+            "false"
+        )
+        ignore_ssl = get_conf_boolean(
+            self.sc,
+            self.ssl_conf,
+            "false"
+        )
+        if is_ssl_enabled:
+            name = self.__class__.__name__
+            if ignore_ssl:
+                print(
```

Comment:
There will be a logger in the next PR. For now, just leaving it as a print() function.

Reply:
you can use `warnings.warn` for now to be consistent with other places